src/memtable.c

```diff
@@ -343,9 +347,8 @@ memtable_context_destroy(platform_heap_id hid, memtable_context *ctxt)
    }
 
    platform_mutex_destroy(&ctxt->incorporation_mutex);
-   platform_free(hid, ctxt->rwlock);
-
-   platform_free(hid, ctxt);
+   platform_free_mem(hid, ctxt->rwlock, ctxt->rwlock_mf_size);
+   platform_free_mem(hid, ctxt, ctxt->mf_size);
```

Comment:
NOTE: Here is an example where we have two memory fragment sizes to be tracked.

1. `rwlock_mf_size` - tracks size of fragment for rw-lock field.
2. `mf_size` - tracks size of fragment for holding `ctxt` structure.

Hence, these two slightly differently named `size`-fields.